spki/src/traits.rs

```diff
@@ -93,3 +96,11 @@ pub trait EncodePublicKey {
         Ok(doc.write_pem_file(path, SubjectPublicKeyInfoRef::PEM_LABEL, line_ending)?)
     }
 }
+
+/// Returns AlgorithmIndentifier associated with the structure. This is mostly useful for Signing
+/// and Validation keys.
+#[cfg(feature = "alloc")]
+pub trait AssociatedAlgorithmIdentifier {
+    /// Returns corresponding AlgorithmIndentifier
+    fn get_algo_id(&self) -> AlgorithmIdentifierOwned;
```

Comment:
`const_oid::AssociatedOid` uses an associated const.

I think both patterns are useful, although naming them seems hard. I was thinking `DynAssociatedOid` for `const_oid`.

There can be a blanket impl which connects them too.

Reply:
Associated const is not really possible since for RSA PSS we have to take `salt_length` into account (which is a `SigningKey` property).

Reply:
@tarcieri see https://github.com/RustCrypto/RSA/pull/278

Reply:
I was saying that there are two patterns that are useful: this one, which takes `&self` and is therefore object safe, and a pattern that uses an associated const.

Though an associated const doesn't work for that, there are many places where `AlgorithmIdentifier` is effectively static, e.g. just a single OID or OID 2-tuple, for example all of the elliptic curve `AlgorithmIdentifier`s.